input/pagecontent/reference.md

```diff
@@ -481,9 +483,9 @@ FHIR represents an element with a choice of datatypes using the style foo[x]. Fo
 
 #### Array Paths using Numerical Indices
 
-FSH uses square-bracketed integers to address elements in arrays. Arrays are referenced using 0-based indices, meaning that the first array element is referenced by `[0]`, the second element is referenced by `[1]`, etc. Arrays with missing elements (gaps in the sequence of indices) are not allowed. If the index is omitted, the first element of the array (`[0]`) is assumed.
+FSH paths SHALL use square-bracketed integers to address elements in arrays. Arrays SHALL be referenced using 0-based indices, meaning that the first array element is referenced by `[0]`, the second element is referenced by `[1]`, etc. Arrays with missing elements (gaps in the sequence of indices) SHALL NOT be allowed. If the index is omitted, the first element of the array (`[0]`) SHALL be assumed.
```

Comment:
I think "SHALL NOT be allowed" is essentially the same as "are not allowed". There's no squishiness in either sentence. IIRC, a null entry in an error will cause an error in the IG Publisher (except in the case of ids and extensions on primitives, when `element` array can contain a null as long as `_element` array contains a value at the corresponding index -- but in FSH we present `element` and `_element` as if the are the same array, so we don't allow null at all).  As for SUSHI, I think we're free to try to handle it gracefully if we want (and we do). That said, I see now that SUSHI doesn't emit a warning or error in this case. We probably ought to.

Reply:
Yeah, I agree they mean the same thing, the "SHALL NOT" just stood out more to me, which made me go back to see what it used to be. I don't think I knew the IG Publisher gives in an error in that case (or maybe I forgot).

Also, it turns out SUSHI does emit a warning when there are empty array items, but it happens outside of the regular export functions that FSH Online uses, so I missed it.

So since this doesn't actually change anything in the FSH Spec, I'm good with leaving it alone.